test/IRGen/generic_metatypes.swift

```diff
@@ -12,13 +12,11 @@
 // RUN: %swift -target x86_64-unknown-linux-gnu -disable-objc-interop -emit-ir -parse-stdlib -primary-file %s | %FileCheck --check-prefix=CHECK --check-prefix=CHECK-64 %s
 
 // REQUIRES: CODEGENERATOR=X86
-
-// FIXME: SR-3591 test is failing after being unintentionally disabled for a while
-// REQUIRES: SR3591
+// REQUIRES: CODEGENERATOR=ARM
```

Comment:
I dislike splitting such tests because it makes it easier for one side to lose coverage. But we don't currently have a better solution for tests that touch multiple code generators. (A new conditional `RUN` command in `lit` would help, something like `RUN REQUIRES CODEGENERATOR=X86: %swift -target x86_64-apple-macosx10.9 …`.)